src/index.ts

```diff
@@ -0,0 +1,52 @@
+import { resolve, join } from 'path';
+import { readJson } from 'fs-extra';
+import * as core from '@actions/core';
+import * as github from '@actions/github';
+import remoteTags from 'remote-git-tags';
+
+const REQUIRED_TAG_FORMAT = '$version';
+
+// eslint-disable-next-line camelcase
+const cloneUrl: string = github.context.payload.repository?.clone_url;
+const tagFormat: string = core.getInput('tag-format');
+
+async function getVersion(): Promise<string> {
+  const packagePath = join(core.getInput('package-directory'), 'package.json');
+  const { version } = await readJson(resolve(packagePath));
```

Comment:
Nit Pick: Instead of `readJson` from fs-extra what are your thoughts on just using fs directly? There is promisified versions of fs functions under `promises` as of node 10, so we can do:

```js
import { promises as fs } from 'fs';

function readJson(jsonFilePath: string):  {
  let firebaseJsonStr: string;
  try {
    const firebaseJsonBuffer = await fs.readFile(jsonFilePath);
    firebaseJsonStr = firebaseJsonBuffer.toString();
  } catch (err) {
    throw new Error(`Error loading ${jsonFilePath}`);
  }
}
```

Reply:
I can use fs directly if you prefer. fs-extra is just so nice and friendly but it does add a little bit of file size (100k) to `dist/index.js`. Not sure if that matters too much with our actions. Let me know if you have a strong preference on that one.

I can try the package path suggestion you mentioned. I was actually looking for something like that so I could just use `join` to put the full path together. Will see how that plays with `ncc`.

Reply:
Not a huge preference on whether to use fs-extra or not, I have just been keeping an eye out for things that can prevent our dependence on external packages when working on core stuff, so it came to mind. Fine to leave as is for now